src/scvi/external/mrvi/_model.py

```diff
@@ -930,10 +951,21 @@ def differential_abundance(
                             stacklevel=2,
                         )
                         continue
-                    rest_log_probs = log_probs_arr.log_probs.loc[{"sample": rest_samples}]
-                    rest_val_log_probs = logsumexp(rest_log_probs, axis=1) - np.log(
-                        rest_log_probs.shape[1]
-                    )
+                    rest_log_probs = log_probs_arr.log_probs.loc[{"sample": rest_samples}].values
+                    if omit_original_sample:
+                        rest_samples_one_hot = np.zeros((adata.n_obs, len(rest_samples)))
+                        for i, sample in enumerate(rest_samples):
+                            rest_samples_one_hot[adata.obs[self.sample_key] == sample, i] = 1
+                        rest_log_probs_no_original = np.where(
+                            rest_samples_one_hot, -np.inf, rest_log_probs
+                        )
+                        rest_val_log_probs = logsumexp(
+                            rest_log_probs_no_original, axis=1
+                        ) - np.log((1 - rest_samples_one_hot).sum(axis=1))
```

Comment:
If this is exact duplicate from above, should we create a private function to mask these out?